### ISSUE TITLE
Each second elapsed on the Serial Monitor actually takes about 2 and a half seconds (maybe a bit more).

### ISSUE THREAD

**Issue by JazzTp** [ZulNs/STM32F1_RTC#3]
Each second elapsed on the Serial Monitor is actually taking about 2 and a half seconds (maybe a bit more).

Possibly related with
+ https://github.com/cbm80amiga/N5110_STM32_RTC/issues/1
+ https://github.com/rogerclarkmelbourne/Arduino_STM32/issues/741

( This issue evolves from https://github.com/ZulNs/STM32F1_RTC/issues/2 )

I'm using Roger Clark's core https://github.com/rogerclarkmelbourne/Arduino_STM32
(I've just updated it, to check if the same occurred with the latest version).

**IDE settings:**
Board "Generic STM32F103C series"
Variant "STM32F103C8 (20k RAM, 64k Flash)"
Upload method "Serial"
CPU Speed(MHz) "72 Mhz (Normal)"
Optimize "Smallest(default)" (other options: Fast, Faster, Fastest, Debug)
Port "/dev/ttyUSB0"
Programmer "USBasp"

**Comment by ZulNs** [comment]
Have you removed PC14 and PC15 pins from its header on the board? Both of these pins are directly connected to 32.768kHz crystal which increase the capacitance effect.

**Comment by JazzTp** [comment]
@ZulNs 

> Have you removed PC14 and PC15 pins from its header on the board? Both of these pins are directly connected to 32.768kHz crystal which increase the capacitance effect.

I'm **truly sorry**, I _had_ read a previous mention [here](https://github.com/rogerclarkmelbourne/Arduino_STM32/issues/741#issuecomment-569897269) of those two pins, but time has come to confess that I don't actually understand.

By removing PC14 and PC15 pins "from its header", do you mean removing from some .h header file? Which one?

I can't figure out how an intervention might be made on the hardware, this is an **STM32F103C8T6** and I don't see anything detachable related with PC14 and PC15... so I guess you mean the software, a header file.

Yet you mentioned capacitance. Would some parasitic capacitance really be able to bring an high Q crystal oscillator to less than 1/2.5 the crystal nominal frequency?

[EDIT: **Those two pins are not being used at the moment and are not connected to anything beyond the perimeter of the breakout board.**]

**Comment by ZulNs** [comment]
I wrote "... from its header on the board". So it related to hardware, not to software. I mean "desoldered".

**Comment by JazzTp** [comment]
**Thanks for replying again.**

OK, you meant "disconnect".

No, I'm not equipped for so tiny soldering/desoldering operations.

(The sketches I linked, however, do not mention any such intervention on the hardware, the ones I've seen being tested in their respective YouTube Videos don't show any, either.)

Aware that a scope probe does introduce further capacitance, out of curiosity, I've anyway given a look at PC15, because I see that it's marked as "OSC32 OUT" on the pinout (PC14 is marked as "OSC32 IN" as you most certainly know).

Rather ugly output, about 350 mV peak-to-peak almost-sine signal at a frequency close to the 32768 kHz, BUT actually on top of a 1.80V peak-to-peak 50 Hz almost-sine wave.

No idea whether that's normal considering the scope probe shouldn't be there, or if this board is defective. I'll buy another one, one of these days, and I'll check the RTC-related sketches again. 

Other signals coming out of this board, e.g. tone to buzzer, result of digitalWrite operations, PWM output, look just fine with no residual 50 Hz AC.
(I understand that it doesn't have much to do with the XTal oscillator, apart being fed by the same power supply. Which BTW is the typical breadboard adapter carrying two linear regulators, fed from a switching power supply giving 9V. I do have a better regulated power supply, I doubt that's the point but I'll give it a try tomorrow, it's 4 a.m. here.)

I also checked PC14: very noisy almost-**square** wave. Measuring after activating the averaging function of the scope (with 256 passes) gives 32.89 kHz, 42 mV peak-to-peak. BUT also riding on top of a (70 mV) 50 Hz ripple. Again, I'm aware that with the probe I'm adding capacitance.


(((
I also have an external DS3231 module here with temperature compensated oscillator, which is working just fine with this board via I²C, I was actually planning to use that one and only wanted to get familiar with the built-in RTC just in case it offered any possibilities that the external DS3231 does not offer. (I can't think of any at the moment, this external module also has an alarm pin and a couple of square wave output pins, one is 32.768 kHz and the other one is allegedly able to provide different frequencies, not seen yet).
)))

**Comment by ZulNs** [comment]
It's not so tiny for desoldering and lifting those pins. It's a normal 2.54mm distance between those pins. Even a standard soldering iron can be used.

**Comment by JazzTp** [comment]
> It's not so tiny for desoldering and lifting those pins. It's a normal 2.54mm distance between those pins. Even a standard soldering iron can be used.

Now I get you. You mean on the perimeter of the breakout board, the perimeter of the PCB, not the perimeter of the MCU chip itself, which is why you wrote  _"remove"_, remove the metal pin from the plastic strip and the PCB hole.

Sure, the breadboard itself might be adding up to 4 pF between adjacent holes.

I'll buy another SM32F103C8T6 board and this time, **_before_** soldering the strip of pins, I'll pull out those two from the plastic strip, or just shorten the strip and put a separate two-pins strip on C13 and VB.

I'll report what happens then. (I might be unable to go buy it today though.)

Thanks again.

**Comment by JazzTp** [comment]
@ZulNs
Sorry for the delay, out of stock in nearby dealers, I ordered **two new boards** on Friday and received them today Tuesday.

**I pulled the metal pins corresponding to C14 and C15 from the strip before soldering.**

**THE BUILT-IN RTC CLOCK IS OK IN BOTH BOARDS** with all the sketches I mentioned.

I didn't test how precise it is in the long run (much probably not as precise as a DS3231), but a quick&dirty check by manual chronometer on one minute is OK.

So I took **the older board** and **removed the pins from the C14 and C15 holes as you suggested** **(*)**.

**THE OLDER BOARD BUILT-IN CLOCK IS OK TOO, NOW** (it has delayed 1 second in 15 minutes but no more than that in the following 45 minutes, when it happened I was passing from one sketch to another, with alarm tests etc., I'll leave it going on battery and check tomorrow).

**SO, YOU'RE THE MAN, _THANK YOU_.**

**Closing this issue, hardware-related and SOLVED.**

---

**(*)** It's certainly an older board despite the fact that I bought it only a few months ago, because R10 is 10k vs. 2k5 on these new ones (I haven't had any problems about that so far anyway, only heard about _possible_ problems mentioned by a blogger who hadn't had any either).

**Just in case anybody comes here and wonders how to remove the pins without damaging the board, if they have already soldered the whole strip, here's how I did it:**

- I gently locked the board in a vise by the pin strips, meaning the PCB portions exceeding the pin strips are resting _on top_ of the vise opposite "jaws", so the vise doesn't need to be awfully tightened
- first, I used a desoldering wick with the soldering iron, to remove as much solder as possible
- then I gently but firmly pulled each pin from below with pliers while warming it from above with the soldering iron, which is most certainly what you were suggesting.

**Comment by JazzTp** [comment]
@ZulNs
Sorry for the delay, out of stock in nearby dealers, I ordered **two new boards** on Friday and received them today Tuesday.

**I pulled the metal pins corresponding to C14 and C15 from the strip before soldering.**

**THE BUILT-IN RTC CLOCK IS OK IN BOTH BOARDS** with all the sketches I mentioned.

I didn't test how precise it is in the long run (much probably not as precise as a DS3231), but a quick&dirty check by manual chronometer on one minute is OK.

So I took **the older board** and **removed the pins from the C14 and C15 holes as you suggested** **(*)**.

**THE OLDER BOARD BUILT-IN CLOCK IS OK TOO, NOW** (it has delayed 1 second in 15 minutes but no more than that in the following 45 minutes, when it happened I was passing from one sketch to another, with alarm tests etc., I'll leave it going on battery and check tomorrow).

**SO, YOU'RE THE MAN, _THANK YOU_.**

**Closing this issue, hardware-related and SOLVED.**

---

**(*)** It's certainly an older board despite the fact that I bought it only a few months ago, because R10 is 10k vs. 2k5 on these new ones (I haven't had any problems about that so far anyway, only heard about _possible_ problems mentioned by a blogger who hadn't had any either).

**Just in case anybody comes here and wonders how to remove the pins without damaging the board, if they have already soldered the whole strip, here's how I did it:**

- I gently locked the board in a vise by the pin strips, meaning the PCB portions exceeding the pin strips are resting _on top_ of the vise opposite "jaws", so the vise doesn't need to be awfully tightened
- first, I used a desoldering wick with the soldering iron, to remove as much solder as possible
- then I gently but firmly pulled each pin from below with pliers while warming it from above with the soldering iron, which is most certainly what you were suggesting.
